posts/2020-11-19-Rust-1.48.md

```diff
@@ -0,0 +1,224 @@
+If you don't have it already, you can [get `rustup`][install] from the
+appropriate page on our website, and check out the [detailed release notes for
+1.48.0][notes] on GitHub.
+
+[install]: https://www.rust-lang.org/tools/install
+[notes]: https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1480-2020-11-19
+
+## What's in 1.48.0 stable
+
+The star of this release is Rustdoc, with a few changes to make writing
+documentation even easier! See the [detailed release notes][notes] to learn
+about other changes not covered by this post.
+
+### Easier linking in rustdoc
+
+Rustdoc, the library documentation tool included in the Rust distribution,
+lets you write documentation in Markdown. This makes it very easy to use, but
+also has some pain points. Let's say that you are writing some documentation
+for some Rust code that looks like this:
+
+```rust
+pub mod foo {
+    pub struct Foo;
+}
+
+pub mod bar {
+    pub struct Bar;
+}
+```
+
+We have two modules, each with a struct inside. Imagine we wanted to use these
+two structs together; we may want to note this in the documentation. So we'd
+write some docs that look like this:
+
+```rust
+pub mod foo {
+    /// Some docs for `Foo`
+    ///
+    /// You may want to use `Foo` with `Bar`.
+    pub struct Foo;
+}
+
+pub mod bar {
+    /// Some docs for `Bar`
+    ///
+    /// You may want to use `Bar` with `Foo`.
+    pub struct Bar;
+}
+```
+
+That's all well and good, but it would be really nice if we could link to these
+other types. That would make it much easier for the users of our library to
+navigate between them in our docs.
+
+The problem here is that Markdown doesn't know anything about Rust, and the
+URLs that rustdoc generates. So what Rust programmers have had to do is write
+those links out manually:
+
+```rust
+pub mod foo {
+    /// Some docs for `Foo`
+    ///
+    /// You may want to use `Foo` with [`Bar`].
+    ///
+    /// [`Bar`]: ../bar/struct.Bar.html
+    pub struct Foo;
+}
+
+pub mod bar {
+    /// Some docs for `Bar`
+    ///
+    /// You may want to use `Bar` with [`Foo`].
+    ///
+    /// [`Foo`]: ../foo/struct.Foo.html
+    pub struct Bar;
+}
+```
+
+Note that we've also had to use relative links, so that this works offline.
+Not only is this process tedious, and error prone, but it's also just wrong
+in places. If we put a `pub use bar::Bar` in our crate root, that would
+re-export `Bar` in our root. Now our links are wrong. But if we fix them,
+then they end up being wrong when we navigate to the `Bar` that lives inside
+the module. You can't actually write these links by hand, and have them all
+be accurate.
+
+In this release, you can use some syntax to let rustdoc know that you're
+trying to link to a type, and it will generate the URLs for you. Here's
+two different examples, based off of our code before:
+
+```rust
+pub mod foo {
+    /// Some docs for `Foo`
+    ///
+    /// You may want to use `Foo` with [Bar](crate::bar::Bar).
+    pub struct Foo;
```

Comment:
It would be nice to show off that scoping makes this really easy most of the time.
```suggestion
    use crate::bar::Bar;
    /// Some docs for `Foo`
    ///
    /// You may want to use `Foo` with [`Bar`].
    pub struct Foo;
```

Thanks for the detailed write-up :) This looks great.

Reply:
I do think this is useful, but I'm not sure if I want to do it here or as another example; that things are consistent all the way through is nice, and I don't want to change *too* much in the same example.